commit 0aeb1def44169cbe7119f26cf10b974a2046142e
Author: Cong Wang <xiyou.wangcong@gmail.com>
Date:   Sat May 25 09:58:01 2019 -0700

    tracing: Make trace_get_fields() global
    
    trace_get_fields() is the only way to read tracepoint fields at
    run time, as their fields are defined at compile-time with macros.
    Make this function visible to all users and it will be used by
    trace event injection code to calculate the size of a tracepoint
    entry.
    Link: http://lkml.kernel.org/r/20190525165802.25944-4-xiyou.wangcong@gmail.com
    
    Cc: Ingo Molnar <mingo@redhat.com>
    Signed-off-by: Cong Wang <xiyou.wangcong@gmail.com>
    Signed-off-by: Steven Rostedt (VMware) <rostedt@goodmis.org>

diff --git a/include/linux/trace_events.h b/include/linux/trace_events.h
index 5c6f2a6c8cd2..5150436783e8 100644
--- a/include/linux/trace_events.h
+++ b/include/linux/trace_events.h
@@ -318,6 +318,14 @@ trace_event_name(struct trace_event_call *call)
 		return call->name;
 }
 
+static inline struct list_head *
+trace_get_fields(struct trace_event_call *event_call)
+{
+	if (!event_call->class->get_fields)
+		return &event_call->class->fields;
+	return event_call->class->get_fields(event_call);
+}
+
 struct trace_array;
 struct trace_subsystem_dir;